pkg/gpu/stats.go

```diff
@@ -84,7 +87,19 @@ func (g *statsGenerator) getOrCreateAggregator(sKey streamKey) *aggregator {
 	}
 
 	if _, ok := g.aggregators[aggKey]; !ok {
-		g.aggregators[aggKey] = newAggregator(g.sysCtx)
+		gpuDevice, err := g.sysCtx.getDeviceByUUID(sKey.gpuUUID)
+		if err != nil {
+			log.Warnf("Error getting device by UUID %s: %s, defaulting to device 0", sKey.gpuUUID, err)
+			gpuDevice = g.sysCtx.gpuDevices[0]
```

Comment:
do we really want that logic? this can mask potential metrics in-accuracies

Reply:
You're right that this probably would be better if we returned an error.

Reply:
Changed to return an error.